doc/release-notes-14954.md

```diff
@@ -0,0 +1,3 @@
+Build system changes
+--------------------
+Python >=3.5 is now required by all aspects of the project. This includes the build systems, test framework, linters and Travis CI. The previously supported minimum (3.4), was EOL in March 2019. See #14954 for more details.
```

Comment:
Not sure if we should mention "travis" as part of the project. Also I didn't explicitly break the build system to not work with 3.4, only the tests. I'd be surprised if anyone ran them, and if someone did with 3.4, they fail anyway, e.g. when `.hex()` is not available.

Reply:
I think having different python version requirements for different parts of the project is confusing, and very hard to enforce, so even if some scripts are compatible with 3.4 (or below) I think it makes sense to be sweeping in this message.

Reply:
Dropped Travis.